### PR TITLE
:sparkles: Milestone needs a responsible user

### PR DESCRIPTION
**Pull request by svenlombaert**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -2698,6 +2698,9 @@ Search projects
             + (object)
                 + due_on: `2018-01-01` (string, required)
                 + name: `Initial setup` (string, required)
+                + responsible_user: (object, required)
+                    + type: `user` (string)
```
Comment (jenssegers) [inline apiary.apib]: Can you make these also required?

Comment (mrtus) [reply apiary.apib]: I thought we would just do `responsible_user_id` for create / update endpoints?

File: apiary.apib [hunk 2]
```diff
@@ -2663,6 +2663,7 @@ Search projects
                 + (object)
                     + type: `milestone` (string)
                     + id: `7c6b5e26-3068-47a6-bf3b-431d7ab7e704` (string)
+                    + responsible_user_id: `e1240972-6cfc-4549-b49c-edda7568cc48` (string, required)
```
Comment (mrnnn) [inline apiary.apib]: Q: I though we always wrapped in objects in responses. @jenssegers ?

Comment (mrtus) [reply apiary.apib]: Yes you are correct @mrnnn 
Good spot!

Comment (jenssegers) [reply apiary.apib]: In responses we return objects, yes. For requests we don't (for now).

**Comment by jenssegers** [comment]
Quick recap:

 - If we pass object id's in requests such as for the `responsible_user`, we allow the user to only pass the id as `responsible_user_id`. Only in the case where the relationship can be of multiple types, we require the full object including the type, such as we do with `customer`.
- For responses we always return both the `id` and `type`, other fields are not allowed on this level. So `responsible_user_id` can not be next to the id and type of a `milestone` field. If the user wants the responsible user, he has to fetch the milestones for that project.

This should be described in the API guidelines: https://github.com/teamleadercrm/api/blob/master/docs/guidelines.md. Since this is missing we should work together in making this more clear for the team. I will go ahead and add a section about the difference between request/responses.

**Pull request merged**